src/components/modebar/manage.js

```diff
@@ -44,7 +45,7 @@ module.exports = function manageModeBar(gd) {
         ].join(' '));
     }
 
-    var customButtons = context.modeBarButtons;
+    var customButtons = extendDeep([], context.modeBarButtons);
```

Comment:
This deep copy appears to be needed in the first case only.
I suggest you revert the change here and move deep copy inside `fillCustomButton` function.
For example:
```js
function fillCustomButton(inButtons) {
  var customButtons = extendDeep([], inButtons);
  ...
```